src/editors/mongoDBDocumentService.ts

```diff
@@ -108,6 +111,32 @@ export default class MongoDBDocumentService {
     }
   }
 
+  _simplifyEJSON(document: Document): Document {
+    for (const [key, item] of Object.entries(document)) {
+      // UUIDs might be represented as {"$uuid": <canonical textual representation of a UUID>} in EJSON
+      // Binary subtypes 3 or 4 are used to represent UUIDs in BSON
+      // But, parsers MUST interpret the $uuid key as BSON Binary subtype 4
+      // For this reason, we are applying this representation for subtype 4 only
+      // see https://github.com/mongodb/specifications/blob/master/source/extended-json.rst#special-rules-for-parsing-uuid-fields
+      if (
+        isObject(item) &&
+        item.hasOwnProperty('$binary') &&
+        item.$binary.subType === '04'
```

Comment:
```suggestion
        Object.prototype.hasOwnProperty.call(item, '$binary') &&
        item.$binary?.subType === '04' &&
        typeof item.$binary.base64 === 'string'
```

You'll almost [never](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty#using_hasownproperty_as_a_property_name) want to use `hasOwnProperty` directly, and you'll want to account for objects like `{ $binary: null }` or `{ $binary: { subType: '04', base64: null } }` because, unfortunately, any of these can be the result of EJSON output (e.g. if they were literal parts of the input)

Reply:
ooh, good to know!